libs/core/src/lib/createScenario.ts

```diff
@@ -13,66 +15,127 @@ import { state } from './state';
 //   T extends { mock: B; options: B['mockOptions'] }[]
 // >
 
-export const createScenario = <T extends Options = Options>(
+type CreateMockOptions = {
+  mockOptions: Record<string, OptionType>;
+  mockTitle: string;
+};
+
+type ScenarioMock = {
+  mock: CreateMockFnReturnType;
+  mockOptions: CreateMockOptions['mockOptions'];
+};
+
+type SetupMocksFn = (
+  options: CreateMockOptions,
+  mockFn: CreateMockMockFn
+) => RestHandler[];
+
+const initializeMocks: SetupMocksFn = (options, mockFn) => {
+  const mockFnReturnValue = mockFn(options.mockOptions);
+  const arrayOfMocks = Array.isArray(mockFnReturnValue)
+    ? mockFnReturnValue
+    : [mockFnReturnValue];
+  return arrayOfMocks;
+};
+
+const initializeManyMocks = ({
+  mocksFromState,
+  createMockOptions,
+  isActive,
+}: {
+  mocksFromState: MocksState[];
+  createMockOptions: CreateMockOptions[];
+  isActive?: boolean;
+}) =>
+  mocksFromState.flatMap(({ createMockBaseArgs: { mockFn } }, index) => {
```

Comment:
Is this a race condition? Could be the test failures have something to do with race conditions.